fatiando/gravmag/transform.py

```diff
@@ -118,6 +125,97 @@ def tga(x, y, data, shape, method='fd'):
     return res
 
 
+def mstde(x, y, data, shape):
+    """
+    Multi-Scale Tilt Depth Estimation
+
+    Implements the method of van Buren (2013).
+
+    An empirical method for attempting to approximate the depth of the source
+    of a magnetic anomaly, based on altering tilt and continuation of the
+    magnetic field intensity.
+
+    This method will also work well on Total Gradient Amplitude, but care must
+    be taken to reduce noise.
+
+    .. warning::
+
+        If the data is not in SI units, the derivatives will be in
+        strange units! I strongly recommend converting the data to SI
+        **before** calculating the derivatives (use one of the unit conversion
+        functions of :mod:`fatiando.utils`). This way the derivative will be in
+        SI units and can be easily converted to what unit you want.
+
+    **References**
+
+    Van Buren, Reece. 2013. “Multi-Scale Tilt Depth Estimation.” MSc Thesis,
+    Johannesberg: University of the Witwatersrand.
+    URI: http://mobile.wiredspace.wits.ac.za/bitstream/handle/10539/14011/MSc_Dissertation_RvanBuren_2013.pdf.
+    """
+
+    tilted_data = tilt(x, y, data, shape)
+    pass
+
+
+def tilt(x, y, data, shape):
+    """
+    Calculates the magnetic tilt, as defined by Miller and Singh (1994):
+
+    tilt(f) = tan^{-1}(\\frac{\\frac{df}{dz}}{\\sqrt{\\frac{df}{dx}^2 + \\frac{df}{dy}^2}})
+
+    Parameters:
+
+    * x, y : 1D-arrays
+        The x and y coordinates of the grid points
+    * data : 1D-array
+        The potential field at the grid points
+    * shape : tuple = (ny, nx)
+        The shape of the grid
+
+    Returns:
+
+    * tilt : 1D-array
+        The tilt angle of the total field.
+
+    """
+    horiz_deriv = thdr(x, y, data, shape)
+    vert_deriv = derivz(x, y, data, shape)
+    tilt_value = vert_deriv/horiz_deriv
+    tilt = numpy.arctan2( tilt_value, tilt_value )
```

Comment:
@mtb-za no need to compute `tilt_value`. You can call `arctan2(vert_deriv, horiz_deriv)`. That's what `arctan2` is for: avoiding the division. If `horiz_deriv` is zero (or close to zero), the division will break `arctan2` will still give you 90º.